various-receipes/component-isolation/test/isolation.test.js

```diff
@@ -116,4 +117,83 @@ describe('/api', () => {
       });
     });
   });
+
+  test('When users service doesn\'t reply within 2 seconds, return 503', async () => {
+    //Arrange
+    nock.cleanAll();
+    nock('http://localhost/user/')
+      .get('/1')
+      .delay(3000)
+      .reply(200);
+    nock('https://mailer.com')
+      .post('/send')
+      .reply(202);
+    const orderToAdd = {
+      userId: 1,
+      productId: 2,
+      mode: 'approved',
+    };
+
+    //Act
+    const response = await request(expressApp).post('/order').send(orderToAdd);
+    
+    //Assert
+    expect(response.status).toBe(503);
+  });
+
+  test('When users service replies with 503, should retry the request', async () => {
```

Comment:
Focus the test on user outcome, not white-box flow. We don't care about the request, rather about the success of the order. Alternative title: "When users service replies with 503 once and retry mechanism is applied, then an order is added successfully "